service/matching/matchingEngine.go

```diff
@@ -115,6 +122,10 @@ type (
 		timeSource           clock.TimeSource
 		// Only set if global namespaces are enabled on the cluster.
 		namespaceReplicationQueue persistence.NamespaceReplicationQueue
+		// Disables concurrent task queue user data updates and replication requests (due to a cassandra limitation)
+		namespaceUpdateLockMap map[string]*namespaceUpdateLocks
```

Comment:
consider using `common/locks.IDMutex` here. it takes some of the boilerplate out of this pattern. you can use separate keys for the update + replication locks

Reply:
I think this is fine TBH